### PR TITLE
roachtest: reassign tests to replication team

### LINKED ISSUE #90125: roachtest: add `OwnerKVReplication` for some tests
We should assign some roachtests to the KV replication team, e.g. `inconsistency`. However, this requires wiring up KV replication as a proper team in `teams.yaml` and elsewhere.

Jira issue: CRDB-20601

### PR DESCRIPTION
**Pull request by erikgrinaker**
**internal/team: add support for team labels**

Previously, roachtest issues were assigned to a GitHub project board,
and Blathers would assign a team label based on that. However, some
teams don't use GitHub project boards.

This patch adds an explicit team label, and makes roachtest set it when
posting issues.

Release note: None
  
**TEAMS: add replication**

Release note: None
  
**roachtest: reassign tests to replication team**

Resolves #90125.

Release note: None

### CODE REVIEW

**Review by renatolabs — APPROVED** [review]
Thanks!

**Review by tbg — COMMENTED** [review]
Note that `p.ExtraLabels` is not consulted to look up an existing issue:

https://github.com/cockroachdb/cockroach/blob/fa899d347221185a272e1dbec6f49a45e5617481/pkg/cmd/internal/issues/issues.go#L296-L305

That might be fine here, just wanted to point it out. The label will only be applied if we're creating an issue.

**Comment by erikgrinaker** [comment]
>Note that p.ExtraLabels is not consulted to look up an existing issue:

I think that's fine. We may have reassigned the existing issue to some other team.

File: pkg/cmd/roachtest/github_test.go [hunk 1]
```diff
@@ -31,6 +31,7 @@ var (
     cockroachdb/rfc-prs: other
   triage_column_id: 0
 cockroachdb/test-eng:
+  label: T-test-eng
```
[the remark this answers is not in the document]
Comment (erikgrinaker) [reply pkg/cmd/roachtest/github_test.go]: Ah, right. Doesn't matter much, since this is just a hardcoded test, but updated.

**Comment by erikgrinaker** [comment]
TFTR!

bors r+

**Comment by erikgrinaker** [comment]
bors retry

**Comment by rafiss** [comment]
Are there still plans to backport e07f78bb4cfa6e0cb77a13dd536fd64251d04755? i'm wondering since that would determine if i can backport https://github.com/cockroachdb/cockroach/pull/92292

**Comment by erikgrinaker** [comment]
Thanks for the nudge, forgot about this.

* https://github.com/cockroachdb/cockroach/pull/92316
* https://github.com/cockroachdb/cockroach/pull/92314

**Pull request merged**